catalog/video/video-mpegts-stream/src/main/java/org/codice/alliance/video/stream/mpegts/plugins/UpdateParent.java

```diff
@@ -0,0 +1,108 @@
+/**
+ * Copyright (c) Codice Foundation
+ * <p/>
+ * This is free software: you can redistribute it and/or modify it under the terms of the GNU Lesser
+ * General Public License as published by the Free Software Foundation, either version 3 of the
+ * License, or any later version.
+ * <p/>
+ * This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without
+ * even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details. A copy of the GNU Lesser General Public License
+ * is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+package org.codice.alliance.video.stream.mpegts.plugins;
+
+import java.util.List;
+import java.util.concurrent.TimeUnit;
+
+import org.codice.alliance.video.stream.mpegts.Context;
+import org.codice.alliance.video.stream.mpegts.framework.FrameworkUtilities;
+import org.codice.alliance.video.stream.mpegts.rollover.RolloverActionException;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import ddf.catalog.data.Metacard;
+import ddf.catalog.operation.UpdateRequest;
+import ddf.catalog.operation.impl.UpdateRequestImpl;
+
+/**
+ * Update the parent metacard by calling an {@link UpdateField} and then by calling the
+ * catalog framework.
+ */
+public class UpdateParent implements FindChildrenStreamEndPlugin.Handler {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(UpdateParent.class);
+
+    private static final long INITIAL_RETY_MILLISECONDS = TimeUnit.SECONDS.toMillis(1);
+
+    private static final long MAX_RETRY_MILLISECONDS = TimeUnit.MINUTES.toMillis(5);
```

Comment:
yes, but it doesn't limit the amount of time a single call to the framework takes. It only comes into play if the update throws an exception (because solr can't find the metacard that was just ingested, there can be a delay)